<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>midi_sequencer</class>
 <widget class="QMainWindow" name="midi_sequencer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>472</width>
    <height>259</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="sizeIncrement">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>midi_sequencer</string>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="2" column="3">
     <widget class="QPushButton" name="stop_button">
      <property name="maximumSize">
       <size>
        <width>80</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>Stop</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="3" colspan="2">
     <widget class="QGroupBox" name="midi_clock_group_box">
      <property name="toolTip">
       <string/>
      </property>
      <property name="title">
       <string>Clock</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="2" alignment="Qt::AlignHCenter">
        <widget class="QDial" name="midi_clock_bpm_dial">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>40</width>
           <height>40</height>
          </size>
         </property>
         <property name="maximum">
          <number>4800</number>
         </property>
         <property name="singleStep">
          <number>1</number>
         </property>
         <property name="value">
          <number>120</number>
         </property>
         <property name="wrapping">
          <bool>false</bool>
         </property>
         <property name="notchTarget">
          <double>40.000000000000000</double>
         </property>
         <property name="notchesVisible">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QComboBox" name="midi_clock_port_box"/>
       </item>
       <item row="2" column="2" alignment="Qt::AlignHCenter">
        <widget class="QLineEdit" name="midi_clock_bpm_edit">
         <property name="minimumSize">
          <size>
           <width>80</width>
           <height>0</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="4">
     <widget class="QLabel" name="sequencer_page_label">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="0" column="3">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="2">
     <widget class="QPushButton" name="play_button">
      <property name="maximumSize">
       <size>
        <width>80</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>Play</string>
      </property>
     </widget>
    </item>
    <item row="4" column="1" rowspan="2" colspan="4">
     <widget class="QGroupBox" name="sequencer_box">
      <property name="title">
       <string>Sequencer</string>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QPushButton" name="bank_button">
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="toolTip">
       <string>+Ctrl: Mute Mode</string>
      </property>
      <property name="text">
       <string>Bank</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0">
     <widget class="QPushButton" name="pattern_button">
      <property name="toolTip">
       <string>+Ctrl: Metronome</string>
      </property>
      <property name="text">
       <string>Pattern</string>
      </property>
     </widget>
    </item>
    <item row="2" column="4">
     <widget class="QPushButton" name="page_button">
      <property name="text">
       <string>Page</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QPushButton" name="track_button">
      <property name="toolTip">
       <string>+Ctrl: Chromatic Mode</string>
      </property>
      <property name="text">
       <string>Track</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>472</width>
     <height>22</height>
    </rect>
   </property>
   <property name="defaultUp">
    <bool>false</bool>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QToolBar" name="main_ool_bar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="status_bar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>midi_clock_port_box</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>midi_sequencer</receiver>
   <slot>select_port(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>22</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>213</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>play_button</sender>
   <signal>clicked()</signal>
   <receiver>midi_sequencer</receiver>
   <slot>start_clock()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>213</x>
     <y>125</y>
    </hint>
    <hint type="destinationlabel">
     <x>137</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stop_button</sender>
   <signal>clicked()</signal>
   <receiver>midi_sequencer</receiver>
   <slot>stop_clock()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>169</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>track_button</sender>
   <signal>clicked()</signal>
   <receiver>midi_sequencer</receiver>
   <slot>change_track()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>714</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pattern_button</sender>
   <signal>clicked()</signal>
   <receiver>midi_sequencer</receiver>
   <slot>change_pattern()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>757</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bank_button</sender>
   <signal>clicked()</signal>
   <receiver>midi_sequencer</receiver>
   <slot>change_bank()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>43</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>748</x>
     <y>243</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>select_port(int)</slot>
  <slot>start_clock()</slot>
  <slot>stop_clock()</slot>
  <slot>set_clock_bpm(int)</slot>
  <slot>set_clock_bpm()</slot>
  <slot>set_clock_bpm_from_dial(int)</slot>
  <slot>change_track()</slot>
  <slot>change_pattern()</slot>
  <slot>change_bank()</slot>
  <slot>sequencer_step_changed()</slot>
 </slots>
</ui>
